### PR TITLE
feat(dpg): property bag for convenience method

### LINKED ISSUE #2476: Generate options parameters types in convenience methods with long parameter lists
As described in [the guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters) and designed in https://github.com/Azure/autorest.csharp/issues/2475.

### PR DESCRIPTION
**Pull request by archerzz**
- add `Parameter.NameExpression` which will provide reference to the parameter under different contexts
- add `LowLevelPropertyBag` which inherits `PropertyBag`
  - corresponding refactoring to `PropertyBag` and `MgmtPropertyBag`
- add a new property `PropertyBag` to `ConvenienceMethod`
- change `OperationMethodChainBuilder`:
  - add a strategy to group method parameters, which is used to switch on/off the grouping behavior
  - modify existing parameter composing logic, change it into 2 stages:
    - first, from protocol parameters to get the expanded convenience method parameters
    - second, group expanded convenience method parameters if count exceeds 5, then compose the final parameter list for convenience method
- refactor `LowLevelClient`, add a code path to specify the grouping strategy for DPG
  - for data plane, use the existing code path of operation method chain builder (e.g. no parameter grouping)
- update tests

resolve #2476

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs [hunk 1]
```diff
@@ -18,6 +18,11 @@ internal record Parameter(string Name, string? Description, CSharpType Type, Con
         public FormattableString? FormattableDescription => Description is null ? (FormattableString?)null : $"{Description}";
         public CSharpAttribute[] Attributes { get; init; } = Array.Empty<CSharpAttribute>();
         public bool IsOptionalInSignature => DefaultValue != null;
+        /// <summary>
+        /// Expression to refer to the parameter under different. It could be name of the parameter, or "optiona.{Name}" depends on the context.
+        /// This is used for expanded parameters.
+        /// </summary>
+        public string NameExpression => (IsPropertyBag ? $"options.{Name.FirstCharToUpperCase()}" : Name);
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs:25]: maybe we could remove the redundant parenthesis

**Comment by archerzz** [comment]
The regen doesn't trigger any change: https://github.com/Azure/azure-sdk-for-net/pull/35071

File: src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs (hunk 1 above)
Comment (archerzz) [reply src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs]: done

**Comment by archerzz** [comment]
Closing since we'll pause on #2476

**Pull request closed without merging**